tests/outbound-wasi-http-v0.2.0-no-perms/test.json5

```diff
@@ -0,0 +1,30 @@
+{
+    "invocations": [
+        {
+            "request": {
+                "path": "/",
+                "headers": [
+                    {
+                        "name": "url",
+                        "value": "http://localhost:%{port=80}"
+                    }
+                ]
+            },
+            "response": {
+                "status": 500,
+                "headers": [
+                    {
+                        "name": "transfer-encoding",
+                        "value": "chunked"
+                    },
```

Comment:
@lann I was a bit surprised by this as the guest is just returning a buffer of bytes. I assume something funny is happening in `wasi-http`'s implementation to cause this.

Reply:
I think this is expected when no `Content-Length` is given.

Reply:
I was able to confirm that indeed, if a `Content-Length` header is give, the `transfer-encoding` header disappears.